pkg/fileutil/purge_test.go

```diff
@@ -47,14 +51,27 @@ func TestPurgeFile(t *testing.T) {
 		}
 		time.Sleep(10 * time.Millisecond)
 	}
-	fnames, err := ReadDir(dir)
-	if err != nil {
-		t.Fatal(err)
-	}
-	wnames := []string{"7.test", "8.test", "9.test"}
-	if !reflect.DeepEqual(fnames, wnames) {
-		t.Errorf("filenames = %v, want %v", fnames, wnames)
+
+	// purge routine should purge 7 out of 10 files and only keep the
+	// 3 most recent ones.
+	// wait for purging for at most 100ms.
+	for i := 0; i < 10; i++ {
+		fnames, err := ReadDir(dir)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(fnames) > 3 {
+			time.Sleep(10 * time.Millisecond)
+			continue
+		}
+
+		wnames := []string{"7.test", "8.test", "9.test"}
+		if !reflect.DeepEqual(fnames, wnames) {
```

Comment:
This still isn't what I meant. Why don't you move the equality check OUTSIDE the loop?